### DISCUSSION TITLE
Copilot saying user not authorized after being accepted for beta testing

### DISCUSSION THREAD

**Discussion by sentwayfarer** [community/community#19553] [Copilot Conversations] [answered]
`[ERROR] [default] [2022-06-27T12:47:48.148Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"`
Reloaded VSCode, reconnected GitHub account, still not working. Anyone else?

**Comment by PurHur** [comment]
ME too. The trail doesnt work

**Reply by davecheney** [reply]
@PurHur there is a problem with your billing details; please contact support and ask them to reset your billing status.

**Reply by jesusugarte10** [reply]
reset please

**Reply by davecheney** [reply]
@jesusugarte10 i'm not seeing any issues with your account. If you are experiencing an unrelated problem, I suggest starting a new discussion thread.

**Reply by PurHur** [reply]
Reset them please. I want to try it without billig information since i have to ask company how we handle the payment issue :1) I opened a ticket

**Comment by the-s-anton** [comment]
Same here, error:

```
[INFO] [auth] [2022-06-27T14:23:31.831Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-06-27T14:23:31.832Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

**Reply by davecheney** [reply]
@the-s-anton there is a problem with your billing details; please contact support and ask them to reset your billing status.

**Comment by 0xmovses** [comment]
same here neovim user

**Reply by davecheney** [reply]
@rvmelkonian i see no requests to api.github.com or copilot-proxy.githubusercontent.com from your account in the last 24 hours. Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Comment by kuroikyu** [comment]
Same error, I was also able to use it during the beta but somehow lost access to it after being widely available, even after I followed the necessary steps.
```
[INFO] [auth] [2022-06-27T19:17:01.756Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-06-27T19:17:01.758Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

PS: I tried uninstalling the extension, logging out of GitHub on VSCode, switching to the Nightly version of the extension... nothing worked.

**Reply by davecheney** [reply]
@kuroikyu  there is a problem with your billing details; please contact support and ask them to reset your billing status.

**Comment by D7EAD** [comment]
Hi, @squoshi!

If you're having issues activating the Github Copilot free 60-day trial on your account, I'd recommend raising a ticket with Github's Support Engineers. You can do so at [this](https://support.github.com/contact/account) link.

**Comment by davecheney** [comment]
@squoshi Please go to https://github.com/github-copilot/tp_signup and complete your signup.

**Comment by Eric-Yangds** [comment]
Copilot testing was acquired earlier this month and used for about two weeks, and today GitHub Copilot could not connect to the server. Extension activation failed: "User not authorized" this problem, how to solve it

**Reply by davecheney** [reply]
@Eric-Yangds it looks like your account details are correct, please go to https://github.com/github-copilot/tp_signup and complete your signup.

**Comment by UtsabDey** [comment]
![image](https://user-images.githubusercontent.com/44144515/176150045-e91286a8-ff74-4327-81a2-568b832c38a5.png)
![image](https://user-images.githubusercontent.com/44144515/176152742-5867ef2c-1f06-4066-90ff-191dd61074bc.png)
![image](https://user-images.githubusercontent.com/44144515/176151206-71d9a7ca-6103-4c02-b418-287537c55b4f.png)

 I have already used copilot for few days, after that I got the error. and try to check the ping and get request time out. I have also apply re-installation process after knowing about the process from community but nothing changed. I have already got the invitation also. Can anyone help me?

**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by UtsabDey** [reply]
> Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

no, it is not possible. cause another one person using same internet service and enjoying copilot

**Reply by davecheney** [reply]
Thank you for confirming. Is it possible this is a firewall or AV issue on your laptop?

**Reply by UtsabDey** [reply]
> Thank you for confirming. Is it possible this is a firewall or AV issue on your laptop?

how to check this issue? I will confirm you right now.

**Reply by UtsabDey** [reply]
> Thank you for confirming. Is it possible this is a firewall or AV issue on your laptop?

no issue with firewall. I checked it.

**Comment by jeffking54** [comment]
I'm getting the same thing, i was in the trial and it was working in VSCode. It suddenly stopped working so I signed up for the trail and put in my Credit Card. Still same error.

```
GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

**Reply by davecheney** [reply]
@jeffking54 please go to https://github.com/settings/copilot and complete the required steps. Copilot wil work once that is completed.

**Comment by 0xmovses** [comment]
Yes still same for me. I have tried the various suggestions in this thread. No firewall or VPN. Github copilot settings all seem fine.

**Comment by PurHur** [comment]
I need this issue to resolve fast. My company will see the 50% drop in productivity if i dont cheat on developing. I will lose my job :(

**Reply by D7EAD** [reply]
Hi.

If you have urgent need for something like Copilot, and can't get access quick enough, I'd suggest taking a look at Tapnine.

**Reply by PurHur** [reply]
I work with both but without copilot i need to press to many keys.

**Comment by Rakeshcv** [comment]
Even I have the same issue. I do have access to copilot but can't use it all of a sudden. It says subscription required.

**Reply by davecheney** [reply]
@Rakeshcv your billing status needs to be reset; please contact support and ask them to reset your billing attempt counter.

**Reply by 0xmovses** [reply]
could this also be the issue with me?

**Reply by davecheney** [reply]
@rvmelkonian i see no requests from your account, I suggest this is a local issue on your workstation. Are you able to try with vscode rather than vim?

**Reply by 0xmovses** [reply]
I just tried it on VScode and it works successfully. so it seems this is an issue the new neovim update. Tim-pope is the maintainer.

**Reply by 0xmovses** [reply]
I think neovim is calling the old api endpoint

**Reply by Rakeshcv** [reply]
@davecheney Thanks for the reply. I have raised the ticket. Thanks again.

**Comment by coliff** [comment]
I have the same issue. I have the plugin installed (I've tried uninstalling and reinstalling and restarting VS Code).

![image](https://user-images.githubusercontent.com/1212885/176355481-0690535c-9440-43c2-ae04-04f66d0f0049.png)
![image](https://user-images.githubusercontent.com/1212885/176355522-43fd7c8a-45b9-4d84-b912-37ecf021c7f7.png)
![image](https://user-images.githubusercontent.com/1212885/176355797-caa12c7f-c76d-408c-a11f-8f090caba96f.png)

I'm eligible to use CoPilot for free but it's not working. The error says 'sign out from your GitHub account using VSCode UI and try again' but how do I do that?

**Reply by davecheney** [reply]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by coliff** [reply]
thanks for super-fast reply! I have this error
![image](https://user-images.githubusercontent.com/1212885/176356362-d67ffa37-5dd9-44fe-b2c7-1183466ec039.png)
![image](https://user-images.githubusercontent.com/1212885/176356499-a8a1aecc-23d8-4f6e-8c24-0283abf71732.png)

And trying to sign in again shows the same error.

**Reply by davecheney** [reply]
That seems unrelated to copilot. That error comes from the vscode settings sync service (I think)

**Reply by coliff** [reply]
finally got it working. thanks @davecheney! The 'Account' button wasn't showing for me in bottom-left so I had to right-click and enable it to show. And then I could sign-out properly and sign back in. Restarted and it's now activated! thanks again :-)

**Comment by 0xmovses** [comment]
I got neovim copilot plugin working simply by uninstalling and reinstalling. I think when I updated my connection was bad and it broke the update. So if anyone else encounters with this neovim, delete the plugin, run `PlugClean` then reinstall.

**Reply by UtsabDey** [reply]
how to plug clean

**Reply by 0xmovses** [reply]
are you using Plug as your plugin manager in neovim?

**Comment by Yan-Daojiang** [comment]
Hi, I have a same error in vscode extension！
```
[INFO] [auth] [2022-07-04T09:17:53.881Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-07-04T09:17:53.886Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

**Comment by sentwayfarer** [comment, marked as answer]
Latest update fixed the issue. Uninstall Copilot, reload VSC, reinstall Copilot, reload VSC, and login.

**Reply by Yan-Daojiang** [reply]
I tried many times, however I still doesn't work.

**Reply by aawor** [reply]
same here

**Reply by GowthamG30** [reply]
I have tried it, but it didn't work.
